### PR TITLE
feat(smart-contract): batch conversion deployer

### PR DESCRIPTION
**Pull request by olivier7delf**
## Description of the changes
- Deploy the batch conversion proxy and setup using xdeployer.
- Delete the batch proxy deployment
- Fix erc20conversion setUp

### CODE REVIEW

File: packages/smart-contracts/scripts-create2/contract-setup/adminTasks.ts [hunk 1]
```diff
@@ -5,8 +5,10 @@ import { BigNumber } from 'ethers';
 
 // Fees: 0.5%
 export const REQUEST_SWAP_FEES = 5;
-// Batch Fees: .3%
-export const BATCH_FEE = 3;
+// Batch fee: temporarily at 0%
```
Comment (olivier7delf) [inline packages/smart-contracts/scripts-create2/contract-setup/adminTasks.ts:8]: `// Batch fee: at 0.3%`

File: packages/smart-contracts/scripts-create2/contract-setup/adminTasks.ts [hunk 2]
```diff
@@ -50,16 +53,32 @@ export const updateRequestSwapFees = async (
...
+};
+
+export const updateBatchPaymentFeeAmountUSDLimit = async (
   contract: any,
-  nonce: number,
   gasPrice: BigNumber,
 ): Promise<void> => {
-  const currentFees = await contract.batchFee();
-  if (currentFees !== BATCH_FEE) {
+  const currentFeeAmountUSDLimit = (await contract.batchFeeAmountUSDLimit()) as number;
```
Comment (leoslr) [inline packages/smart-contracts/scripts-create2/contract-setup/adminTasks.ts:72]: Same here

**Review by leoslr — APPROVED** [review]

File: packages/smart-contracts/scripts-create2/contract-setup/setupErc20ConversionProxy.ts [hunk 3]
```diff
@@ -6,6 +6,8 @@ import {
   updatePaymentFeeProxyAddress,
 } from './adminTasks';
 
+const ERC20ConversionVersion = '0.1.2';
```
Comment (olivier7delf) [inline packages/smart-contracts/scripts-create2/contract-setup/setupErc20ConversionProxy.ts:9]: Needed to use the right RC20Conversion ABI file

File: packages/smart-contracts/scripts-create2/contract-setup/setupErc20ConversionProxy.ts [hunk 4]
```diff
@@ -18,14 +20,13 @@ export const setupErc20ConversionProxy = async (
   // Setup contract parameters
   const Erc20ConversionProxyContract = new hre.ethers.Contract(
     contractAddress,
-    erc20ConversionProxy.getContractAbi(),
+    erc20ConversionProxy.getContractAbi(ERC20ConversionVersion),
```
Comment (alexandre-abrioux) [inline packages/smart-contracts/scripts-create2/contract-setup/setupErc20ConversionProxy.ts:23]: Why are we not using the latest version? Just curious

File: packages/smart-contracts/scripts-create2/contract-setup/adminTasks.ts [hunk 5]
```diff
@@ -8,21 +8,23 @@ import { HardhatRuntimeEnvironmentExtended } from '../types';
 // Fees: 0.5%
 export const REQUEST_SWAP_FEES = 5;
 // Batch Fees: .3%
-export const BATCH_FEE = 3;
+export const BATCH_FEE = BigNumber.from(30);
```
[the remark this answers is not in the document]
Comment (olivier7delf) [reply packages/smart-contracts/scripts-create2/contract-setup/adminTasks.ts:11]: Yes it is for precision :)

File: packages/smart-contracts/scripts-create2/contract-setup/setupErc20ConversionProxy.ts (hunk 4 above)
Comment (olivier7delf) [reply packages/smart-contracts/scripts-create2/contract-setup/setupErc20ConversionProxy.ts:23]: Within erc20Conversion's index, most of the addresses are registered within version "0.0.1", but their real ABI is "0.1.2".
This contract indexes is quite confusing. If someone want to update its setUp, he may have to choose the version...

**Pull request merged**